src/test/java/seedu/address/logic/commands/EditCommandTest.java

```diff
@@ -110,46 +119,21 @@ public void execute_duplicateModuleFilteredList_failure() {
         showModuleAtIndex(model, INDEX_FIRST_MODULE);
 
         // edit module in filtered list into a duplicate in GradPad
-        Module moduleInList = model.getGradPad().getModuleList().get(INDEX_SECOND_MODULE.getZeroBased());
-        EditCommand editCommand = new EditCommand(INDEX_FIRST_MODULE,
+        Module moduleInList = model.getGradPad().getModuleList().stream()
+            .filter(x -> x.getModuleCode().equals(CODE_SECOND_MODULE)).findFirst().get();
+        EditCommand editCommand = new EditCommand(CODE_FIRST_MODULE,
                 new EditModuleDescriptorBuilder(moduleInList).build());
 
         assertCommandFailure(editCommand, model, EditCommand.MESSAGE_DUPLICATE_MODULE);
     }
 
-    @Test
-    public void execute_invalidModuleIndexUnfilteredList_failure() {
```

Comment:
Should we include negative tests for invalid module codes instead? i.e when a user tries to edit a module that doesn't exist in `Completed Modules`

Reply:
ok will include!